10up/ElasticPress
# Fix/1486 auto query

### Description of the Change

This PR moves the autosuggest query generation from the javascript into PHP in order to ensure query consistency and avoid development overhead when integrating new features.
This addresses also #1486 and resolves the main issue. 

It does not yet address the edge case to disable the search feature all-together if no fields have been selected.

### Verification Process

- Install EP / Core WP stock 2019 theme / default content
- Activate / configure plugin and ensure autosuggest is enabled
- navigate to Manage Search Fields & Weighting
- Disable all search fields except title for post type post
- create a post with title "1486-Test post" and publish it
- reload site and type 1486 into search box
- "post 1486-Test post" should appear in autosuggest dropdown
- Create a page with title "1486-Test page" and publish it
- reload site and type 1486 into search box
- only post "1486-Test post" should appear in autosuggest dropdown
- navigate to Manage Search Fields & Weighting
- Enable post title for post type page
- reload site and type 1486 into search box
- both post "1486-Test post" and "1486-Test page" should appear in autosuggest dropdown
- try various variations of this with different fields / post types and confirm expected results

### Applicable Issues

#1486 
possibly #1386

### Changelog Entry

This PR moves the autosuggest query generation from the javascript into PHP in order to ensure query consistency and avoid development overhead when integrating new features.
This addresses also #1486 and resolves the main issue.
